src/data/weather.ts

```diff
@@ -523,3 +529,61 @@ export const isForecastHourly = (
 
   return undefined;
 };
+
+export const Forecast_type = (
+  forecast?: ForecastAttribute[],
+  forecast_daily?: ForecastAttribute[],
+  forecast_hourly?: ForecastAttribute[],
+  forecast_twice_daily?: ForecastAttribute[],
+  forecast_type?: string
+): ForecastAttribute[] | undefined => {
+  if (
+    forecast_type === "daily" &&
+    forecast_daily?.length &&
+    forecast_daily?.length > 2
+  ) {
+    return forecast_daily;
+  }
+  if (
+    forecast_type === "hourly" &&
+    forecast_hourly?.length &&
+    forecast_hourly?.length > 2
+  ) {
+    return forecast_hourly;
+  }
+  if (
+    forecast_type === "twice_daily" &&
+    forecast_twice_daily?.length &&
+    forecast_twice_daily?.length > 2
+  ) {
+    return forecast_twice_daily;
+  }
+  if (
+    (forecast_type === "legacy" || forecast_type === undefined) &&
+    forecast?.length &&
+    forecast?.length > 2
+  ) {
+    return forecast;
+  }
+
+  return undefined;
+};
+
+export const Forecast_type_undefined = (
```

Comment:
Can we combine this function with the one above? as `forecast_type` is already optional?